plugins/org.eclipse.fordiac.ide.deployment/src/org/eclipse/fordiac/ide/deployment/data/ResourceDeploymentData.java

```diff
@@ -137,18 +139,72 @@ public static String getMappedParentName(final FBNetworkElement fbnElement) {
 
 	private void addSubAppParams(final SubApp subApp, final Deque<SubApp> subAppHierarchy, final String prefix)
 			throws DeploymentException {
+		// follow the connection through the subapp hierachy until there is a loose one
 		for (final VarDeclaration dataInput : subApp.getInterface().getInputVars()) {
-			final Function<VarDeclaration, String> dataInputValue = DeploymentHelper
-					.getVariableValueRetargetable(dataInput, subApp);
-			if (dataInputValue != null) {
-				for (final ConDeploymentDest destData : getSubappInterfaceconnections(subAppHierarchy, prefix,
-						dataInput)) {
-					final VarDeclaration destVar = (VarDeclaration) destData.destination;
-					final String destDataValue = dataInputValue.apply(destVar);
-					params.add(new ParameterData(destDataValue, destData.prefix, destVar));
+
+			if (useSubAppInitialValue(subAppHierarchy, subApp, dataInput)) {
+				final Function<VarDeclaration, String> dataInputValue = DeploymentHelper
+						.getVariableValueRetargetable(dataInput, subApp);
+				if (dataInputValue != null) {
+					for (final ConDeploymentDest destData : getSubappInterfaceconnections(subAppHierarchy, prefix,
+							dataInput)) {
+						final VarDeclaration destVar = (VarDeclaration) destData.destination;
+						final String destDataValue = dataInputValue.apply(destVar);
+						params.add(new ParameterData(destDataValue, destData.prefix, destVar));
+					}
+				}
+			}
+		}
+	}
+
+	private static boolean useSubAppInitialValue(final Deque<SubApp> subAppHierachy, final SubApp subApp,
+			final VarDeclaration varDec) {
+		final IInterfaceElement ie = subApp.getInterfaceElement(varDec.getName());
+		if (!hasInputConnections(ie)) {
+			return DeploymentHelper.hasTypeOrInstanceInitialValue(subApp, varDec);
```

Comment:
I'm not sure if hsyTypeirInstanceInitalValue is doing what it should do here. We need to check if either the instance of the type has an value. The method checks if the pin has a value or if the subapp is typed. This is something different.